lib/ChangeDueDateDialog/ChangeDueDate.js

```diff
@@ -159,17 +160,27 @@ class ChangeDueDate extends React.Component {
   }
 
   changeDueDate() {
-    const loans = this.getUpdatedLoans();
-    const promises = loans
+    this.props.setProcessing(true);
+
+    this.getUpdatedLoans()
       // Adding a catch block to the individual loan promises
       // traps the errors that the server returns if there's an
       // 'unproccessable entity', e.g. because the item is declared
       // lost. This allows Promise.all to resolve *all* the promises
       // in the array without stopping at the first error.
-      .map(loan => this.performLoanChange(loan).catch(e => e));
-
-    Promise.all(promises)
-      .then(results => this.props.onDueDateChanged(results));
+      .map(loan => this.performLoanChange(loan).catch(e => e))
+
+      // Previously, this function used Promise.all() to execute these loan changes
+      // in parallel. But this was overloading the backend, where the transactions involved
+      // are complex, and causing many of the requests to time out with 500 errors. We're
+      // switching to sequential execution here in an attempt to keep the backend happy.
+      .reduce(
```

Comment:
@Baroquem I'm glad this works! It maybe just me but it's very hard to follow what is happening here. Could this be replaced with just a simple for loop and async/await? 

For example:

````js
const loans = this.getUpdatedLoans();

for (let i = 0; i < loans.length; i++) {
   try {
     const result = await this.performLoanChange(loans[i]);
     results.push(result);
   }
   catch (e) {}
}

this.props.onDueDateChanged(results); 

````

Reply:
@mkuklis, thanks for the suggestion. It is easier to read! I like it.